src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -2068,18 +2057,19 @@ private ExpressionEntry HandleCall(MethodDesc callee, MethodSignature signature,
             }
         }
 
-        private LLVMValueRef HandleCall(MethodDesc callee, MethodSignature signature, StackEntry[] argumentValues,
-            ILOpcode opcode, TypeDesc constrainedType, LLVMValueRef calliTarget, int offset, LLVMValueRef baseShadowStack, LLVMBuilderRef builder, bool needsReturnSlot,
-            LLVMValueRef castReturnAddress)
+        void AddDependencyForMethodCall(MethodDesc callee, ILOpcode opcode)
         {
-            if (opcode == ILOpcode.callvirt && callee.IsVirtual)
-            {
-                AddVirtualMethodReference(callee);
-            }
-            else if (callee != null)
+            if (callee != null && !(opcode == ILOpcode.callvirt && callee.IsVirtual))
             {
                 AddMethodReference(callee);
             }
+        }
+
+        private LLVMValueRef HandleCall(MethodDesc callee, MethodSignature signature, StackEntry[] argumentValues,
+            ILOpcode opcode, TypeDesc constrainedType, LLVMValueRef calliTarget, int offset, LLVMValueRef baseShadowStack, LLVMBuilderRef builder, bool needsReturnSlot,
+            LLVMValueRef castReturnAddress)
+        {
+            AddDependencyForMethodCall(callee, opcode);
```

Comment:
Do we really need this line? It kind of looks like `LLVMFunctionForMethod` will already add the necessary dependencies (and if it's not doing that for all cases, we should probably fix it there).

Reply:
Well spotted, it does indeed appear to be completely covered there.